### PR TITLE
chore: add pool client with only modifications for building

### PR DESCRIPTION
**Pull request by daywiss**
Signed-off-by: David <david@umaproject.org>

Migrates v1 pool client over to v2. Imports stuff from uma/sdk as well. code hasnt changed other than that. tests pass.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -2,8 +2,12 @@ module.exports = {
   "extends": [
     "react-app",
     "prettier/@typescript-eslint",
-    "plugin:prettier/recommended"
+    "plugin:prettier/recommended",
+    "eslint:recommended",
   ],
+  rules: {
+    "prettier/prettier": ["warn"],
```
Comment (daywiss) [inline .eslintrc.js:9]: this was giving me an error on somethign that isnt an error. it was erroring where i use `import type`

```
/home/david/src/across/sdk-v2/src/pool/bridgePool.ts                                                            
 5:14  warning  Parsing error: '=' expected  prettier/prettier  
```

File: jest-e2e-config.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "testMatch": ["**/*.e2e.ts"]
```
Comment (daywiss) [inline jest-e2e-config.json:2]: this is to run jest e2e tests

File: package.json [hunk 3]
```diff
@@ -39,14 +41,17 @@
     }
   ],
   "devDependencies": {
+    "@eth-optimism/contracts": "^0.5.18",
```
Comment (daywiss) [inline package.json:44]: this was required to import `@uma/sdk`

File: src/utils.ts [hunk 4]
```diff
@@ -56,3 +61,131 @@ export function max(a: BigNumberish, b: BigNumberish): BN {
...
+  periodsElapsed: Decimalish,
+  periodsPerYear: Decimalish
+): string => {
+  return new Decimal(endAmount)
+    .sub(startAmount)
+    .div(startAmount)
+    .mul(periodsPerYear)
+    .div(periodsElapsed)
+    .toString();
+};
```
Comment (daywiss) [inline src/utils.ts:191]: these are all kind of specific to pool client but decided might as well just put it in this util file

File: .eslintrc.js [hunk 5]
```diff
@@ -2,8 +2,12 @@ module.exports = {
   "extends": [
     "react-app",
     "prettier/@typescript-eslint",
-    "plugin:prettier/recommended"
+    "plugin:prettier/recommended",
```
Comment (amateima) [inline .eslintrc.js:5]: I think `plugin:prettier/recommended` should always be used as the last item in the array to make sure all rules that conflict with Prettier are overwritten

**Review by amateima — APPROVED** [review]

File: .eslintrc.js (hunk 5 above)
Comment (daywiss) [reply .eslintrc.js:5]: i see ok, i can actually remove the one i added, i was trying to get rid of the error i was getting

File: src/pool/bridgePool.e2e.ts [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+import { Client, Provider, PoolEventState } from "./bridgePool";
+import { ethers, BigNumber } from "ethers";
+import { TransactionReceipt } from "@ethersproject/abstract-provider";
+import assert from "assert";
+import set from "lodash/set";
+import get from "lodash/get";
+
+const { bridgePool } = uma.clients;
+
+dotenv.config();
```
Comment (amateima) [inline src/pool/bridgePool.e2e.ts:12]: I'm just thinking, if each "sub-package" will require a different set of env variables, would be necessary to have one .env file for each 'sub-package' or keep a single .env file for the whole project?

Comment (daywiss) [reply src/pool/bridgePool.e2e.ts:12]: hmm, good question, i would just keep a single env but open to a different way

**Pull request merged**